crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -1353,8 +1344,58 @@ impl<InstalledPackages: InstalledPackagesProvider> ResolverState<InstalledPackag
                         }
                     }
                 }
+
                 true
             })
+            .flat_map(move |requirement| {
+                iter::once(requirement).chain(
+                    self.constraints
+                        .get(&requirement.name)
+                        .into_iter()
+                        .flatten()
+                        .filter(move |constraint| {
+                            // If the constraint would not be selected with any Python version
+                            // supported by the root, skip it.
+                            if !satisfies_requires_python(self.requires_python.as_ref(), constraint) {
+                                trace!(
+                                    "skipping {constraint} because of Requires-Python {requires_python}",
+                                    // OK because this filter only applies when there is a present
+                                    // Requires-Python specifier.
+                                    requires_python = self.requires_python.as_ref().unwrap()
+                                );
+                                return false;
+                            }
+
+                            // If we're in universal mode, `fork_markers` might correspond to a
+                            // non-trivial marker expression that provoked the resolver to fork.
+                            // In that case, we should ignore any dependency that cannot possibly
+                            // satisfy the markers that provoked the fork.
+                            if !possible_to_satisfy_markers(markers, constraint) {
+                                trace!("skipping {constraint} because of context resolver markers {markers}");
+                                return false;
+                            }
+
+                            // If the constraint isn't relevant for the current platform, skip it.
+                            match extra {
+                                Some(source_extra) => {
+                                    if !constraint.evaluate_markers(
+                                        self.markers.as_ref(),
+                                        std::slice::from_ref(source_extra),
+                                    ) {
+                                        return false;
+                                    }
+                                }
+                                None => {
+                                    if !constraint.evaluate_markers(self.markers.as_ref(), &[]) {
+                                        return false;
+                                    }
+                                }
+                            }
```

Comment:
This is almost the same as above but not quite (the filtering on extras is different), just as it was prior to the refactor (intentionally so).